### DISCUSSION TITLE
CC Tweaked + Applied Energistics 2 - Access to more detailed data

### DISCUSSION THREAD

**Discussion by kid2407** [cc-tweaked/CC-Tweaked#1289] [Q&A] [answered]
What I want: Access the data an ME System has, e.g. items stored, stored power, crafting cpus/jobs etc.

I have learned about the existence of Plethora, though it isn't really clear to me what options it provides and which not.

From what I have gathered in the plethora docs: This is not included in what the mod provides, just a basic-ish "this is a ae2 block" data.

If not covered elsewhere, where would I need to hook in with an own sort of "cc extras for ae2 mod" to add methods a cc computer can see / use?

Edit: Found this old issue but couldn't really grasp the info I needed from that: https://github.com/cc-tweaked/CC-Tweaked/issues/382

**Comment by SquidDev** [comment, marked as answer]
Plethora should support what you're looking for, it's just not documented on the main site (that only includes vanilla docs). There's a list of [all available methods on this page](https://plethora.madefor.cc/docs.html#targeted-methods-appeng.api.networking.IGrid). It's been a while since I looked at any of this, but I'm fairly sure it can do everything you're looking for, though you may need to fiddle a bit.

Worth noting that Plethora is 1.12 only. There are some AE integration mods for later versions of MC - I can think of [Storage for ComputerCraft](https://www.curseforge.com/minecraft/mc-mods/storage-for-computercraft) and [Advanced Peripherals](https://advancedperipherals.madefor.cc/) off the top of my head.

**Reply by kid2407** [reply]
Those are some nice options, given I'm on 1.18.2 in this case it seems Advanced Peripherals offers what I want.

**Reply by SirEdvin** [reply]
You also can use [UnlimitedPeirpheralWorks](https://www.curseforge.com/minecraft/mc-mods/unlimitedperipheralworks/) for Fabric 1.18 and 1.19.2

**Reply by kid2407** [reply]
> You also can use [UnlimitedPeirpheralWorks](https://www.curseforge.com/minecraft/mc-mods/unlimitedperipheralworks/) for Fabric 1.18 and 1.19.2

Sadly I'm using Forge so Fabric mods not really being an option here.
